src/xrc/src/forex.rs

```diff
@@ -1089,29 +1141,29 @@ mod test {
 
         assert!(matches!(
             store.get(1234, 1234, "EUR", USD),
-            Ok(rate) if rate.rates == vec![1_000_000_000] && rate.base_asset_num_received_rates == 5,
+            Ok(rate) if rate.rates == vec![1_000_000_000, 1_000_000_000, 1_000_000_000, 1_000_000_000, 1_000_000_000] && rate.base_asset_num_received_rates == 5,
         ));
         assert!(matches!(
             store.get(1234, 1234, "SGD", USD),
-            Ok(rate) if rate.rates == vec![1_000_000_000] && rate.base_asset_num_received_rates == 5,
+            Ok(rate) if rate.rates == vec![1_000_000_000, 1_000_000_000, 1_000_000_000, 1_000_000_000, 1_000_000_000] && rate.base_asset_num_received_rates == 5,
         ));
         assert!(matches!(
             store.get(1234, 1234, "CHF", USD),
-            Ok(rate) if rate.rates == vec![1_000_000_000] && rate.base_asset_num_received_rates == 5,
+            Ok(rate) if rate.rates == vec![1_000_000_000, 1_000_000_000, 1_000_000_000, 1_000_000_000, 1_000_000_000] && rate.base_asset_num_received_rates == 5,
         ));
         assert!(matches!(
             store.get(1234, 1234, "GBP", USD),
-            Ok(rate) if rate.rates == vec![1_000_000_000] && rate.base_asset_num_received_rates == 2,
+            Ok(rate) if rate.rates == vec![1_000_000_000, 1_000_000_000, 1_000_000_000, 1_000_000_000, 1_000_000_000, 1_000_000_000] && rate.base_asset_num_received_rates == 6,
         ));
 
         assert!(matches!(
             store.get(1234, 1234, USD, "CAD"),
-            Ok(rate) if rate.rates == vec![400_000_000] && rate.base_asset_num_received_rates == 2,
+            Ok(rate) if rate.rates == vec![400_000_000, 400_000_000] && rate.base_asset_num_received_rates == 2,
         ));
 
         assert!(matches!(
             store.get(1234, 1234, "CHF", "EUR"),
-            Ok(rate) if rate.rates == vec![1_000_000_000] && rate.base_asset_num_received_rates == 5 && rate.base_asset.symbol == "CHF" && rate.quote_asset.symbol == "EUR",
+            Ok(rate) if rate.rates.len() == 25 && rate.rates.iter().all(|value| *value == 1_000_000_000) && rate.base_asset_num_received_rates == 5 && rate.base_asset.symbol == "CHF" && rate.quote_asset.symbol == "EUR",
```

Comment:
Nice iteration trick!

Why don't you use it for the other (shorter) vectors? Do you think it is easier to read when providing the full vectors?

Reply:
Done